src/language_features/code_lens.rs

```diff
@@ -32,10 +32,54 @@ pub fn text_document_code_lens(meta: EditorMeta, ctx: &mut Context) {
         partial_result_params: PartialResultParams::default(),
     };
     ctx.call::<CodeLensRequest, _>(meta, req_params, |ctx: &mut Context, meta, result| {
-        editor_code_lens(meta, result, ctx)
+        editor_code_lens(meta.clone(), result.clone(), ctx);
+        inlay_hints_from_editor_code_lens(meta, result.unwrap_or_default(), ctx)
```

Comment:
If we merge the additions into `editor_code_lens`, we get rid of some boilerplate and the clones here

Reply:
I'm usually in favor of keeping _some_ boilerplate in favor of increasing code readability and separation of concerns. It's not really going to increase the speed of the LSP as io would probably dominate everything.

Let's leave it the same for now while this PR works itself towards completion. You can take a call again later.